### PR TITLE
Basic (high level) tests for proposal with GPT

### PR DESCRIPTION
**Pull request by ancorgs**
It includes some refactoring of the tests, review by commit to see that.

### CODE REVIEW

**Comment by aschnell** [comment]
The rspec files repeat identical blocks multiple times, e.g. ```context "with a separate home" ... context "without separate home" ... ```.

File: test/proposal_test.rb [hunk 1]
```diff
@@ -21,326 +21,74 @@
...
-          let(:separate_home) { false }
-          include_examples "proposed layout"
-        end
-      end
+    it "reuses suitable swap partitions" do
+      proposal.propose
+      sda3 = proposal.devices.partitions.with(name: "/dev/sda3").first
+      expect(sda3).to match_fields(
+        mountpoint: "swap",
+        uuid:       "33333333-3333-3333-3333-33333333",
```
Comment (aschnell) [inline test/proposal_test.rb]: I assume you made up that UUID: In general UUIDs follow versions and rules, see https://en.wikipedia.org/wiki/Universally_unique_identifier. The one here might be invalid.

File: test/support/proposal_context.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  before do
+    fake_scenario(scenario)
+
+    allow(Y2Storage::DiskAnalyzer).to receive(:new).and_return disk_analyzer
+    allow(disk_analyzer).to receive(:windows_partition?) do |partition|
+      !!(partition.filesystem.label =~ /indows/)
+    end
+
+    allow_any_instance_of(::Storage::Filesystem).to receive(:detect_resize_info)
+      .and_return(resize_info)
```
Comment (aschnell) [inline test/support/proposal_context.rb]: Setting the resize info in the library would increase the usefulness of the tests in that the real API is used.

File: test/proposal_test.rb (hunk 1 above)
Comment (ancorgs) [inline test/proposal_test.rb]: Well, this test is not new, just re-indented. That being said, the idea of the test if to prove than the UUIDS are reused, whatever their are. IMHO, using obviously made-up UUIDs for that purpose makes it more obvious and more readable.

**Comment by ancorgs** [comment]
I will try to reduce repetition of contexts

**Comment by ancorgs** [comment]
Replaced by #130

**Pull request closed without merging**